sdk/core/azure-core/src/main/java/com/azure/core/credential/TokenRefreshOptions.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.credential;
+
+import java.time.Duration;
+
+/**
+ * The options to configure the token refresh behavior.
+ */
+public class TokenRefreshOptions {
+    private static final Duration DEFAULT_RETRY_TIMEOUT = Duration.ofSeconds(30);
+    private static final Duration DEFAULT_OFFSET = Duration.ofMinutes(2);
+
+    private Duration retryTimeout = DEFAULT_RETRY_TIMEOUT;
+    private Duration offset = DEFAULT_OFFSET;
+
+    /**
+     * Returns a Duration value representing the amount of time to wait before retrying a token refresh. This is to
+     * prevent sending too many requests to the authentication service.
+     *
+     * @return the duration value representing the amount of time to wait before retrying a token refresh
+     */
+    public Duration getRetryTimeout() {
```

Comment:
I worry that this property is very complex  to use. TokenCredentialOptions already have retry options, and this adds another level of complexity to that. If we make it configurable users might have to understand the interrelated nature of this and the retry options. Perhaps we could defer allowing this to be configurable? If not I don't think that `RetryTimeout` really describes what this property is used for.

Reply:
On `IdentityClientOptions` this is called "tokenRefreshRetryTimeout()`. The name is only simplified as you suggested in `TokenRefreshOptions`. I agreed to make the change considering this is in the context of the `TokenRefreshOptions` class.

For .NET if you already have a `TokenCredentialsOptions` it could make more sense to move this inside `TokenCredentialOptions`. It could live there as a `RefreshOptions` property bag, or flattened as `RefreshTimeout` and `RefreshOffset` properties.

Reply:
On second thought, I'll make the case for this class to be immutable as an exception submitted to Java architects. That will make sure these are only modifiable from a subclass.

Reply:
@schaabs Considering that in the current Java implementation, the timeout only applies to error retries, but also refreshes after a previous successful refresh, this may be better called "getTimeout()" in `TokenRefreshOptions.java`, or `TokenRefreshTimeout` in other places like `TokenCredentialOptions.cs`. The scenario happens if a user specifies a "tokenRefreshOffset" that's equal to or longer than the token valid time, it prevents the cache from proactive refreshing on every single request.